underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/SurveyFilterBuilder.java

```diff
@@ -0,0 +1,279 @@
+package bio.terra.tanagra.filterbuilder.impl.core;
+
+import static bio.terra.tanagra.filterbuilder.impl.core.utils.AttributeSchemaUtils.IGNORED_ATTRIBUTE_NAME_UI_USE_ONLY;
+import static bio.terra.tanagra.utils.ProtobufUtils.deserializeFromJsonOrProtoBytes;
+
+import bio.terra.tanagra.api.filter.BooleanAndOrFilter;
+import bio.terra.tanagra.api.filter.EntityFilter;
+import bio.terra.tanagra.api.filter.PrimaryWithCriteriaFilter;
+import bio.terra.tanagra.api.shared.Literal;
+import bio.terra.tanagra.exception.InvalidQueryException;
+import bio.terra.tanagra.exception.SystemException;
+import bio.terra.tanagra.filterbuilder.EntityOutput;
+import bio.terra.tanagra.filterbuilder.FilterBuilder;
+import bio.terra.tanagra.filterbuilder.impl.core.utils.AttributeSchemaUtils;
+import bio.terra.tanagra.filterbuilder.impl.core.utils.EntityGroupFilterUtils;
+import bio.terra.tanagra.filterbuilder.impl.core.utils.GroupByCountSchemaUtils;
+import bio.terra.tanagra.proto.criteriaselector.configschema.CFSurvey;
+import bio.terra.tanagra.proto.criteriaselector.configschema.CFUnhintedValue;
+import bio.terra.tanagra.proto.criteriaselector.dataschema.DTSurvey;
+import bio.terra.tanagra.proto.criteriaselector.dataschema.DTUnhintedValue;
+import bio.terra.tanagra.underlay.Underlay;
+import bio.terra.tanagra.underlay.entitymodel.Attribute;
+import bio.terra.tanagra.underlay.entitymodel.Entity;
+import bio.terra.tanagra.underlay.entitymodel.entitygroup.CriteriaOccurrence;
+import bio.terra.tanagra.underlay.entitymodel.entitygroup.EntityGroup;
+import bio.terra.tanagra.underlay.uiplugin.CriteriaSelector;
+import bio.terra.tanagra.underlay.uiplugin.SelectionData;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+@SuppressFBWarnings(
+    value = "NP_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD",
+    justification = "The config and data objects are deserialized by Jackson.")
+public class SurveyFilterBuilder extends FilterBuilder {
```

Comment:
Chunks of this class are pretty similar to the existing `EntityGroupFilterBuilder`. I think it would be better to move those chunks into a shared base class or a utility class. Perhaps using a generic type would be the least amount of up-front work (e.g. `<T> = <DTEntityGroupEntityGroup> or <DTSurvey.Survey>`).